hutch_python/startup_script.py

```diff
@@ -19,10 +19,27 @@ def _configure_elog_poster():
     from hutch_python.utils import safe_load
 
     with safe_load('ELogPoster'):
-        # RE, ELog will already exist by now
-        elog = globals().get('elog', None)
-        RE = globals().get('RE', None)
-        assert elog is not None
+        try:
+            from bluesky.run_engine import RunEngine
+            from elog import HutchELog
+        except ImportError:
+            raise RuntimeError("A required module is missing, skip ElogPoster")
+
+        # RE, elog will already exist by now, if not we fail and skip.
+        # Some effort to give better error message for the logs.
+        try:
+            elog = globals()['elog']
+        except KeyError:
+            raise RuntimeError("elog not loaded, skip ElogPoster")
+        try:
+            RE = globals()['RE']
+        except KeyError:
+            raise RuntimeError("RE not loaded, skip ElogPoster")
+        # Even if they exist, things can go wrong if the user accidentally clobbers the name
+        if not isinstance(elog, HutchELog):
+            raise RuntimeError("elog replaced, skip ELogPoster")
```

Comment:
I guess this is something the hint banner could also fall prey to, if someone overwrites an existing namespace global.  

I think it's maybe a small miracle that we haven't hit upon a case where a hutch overwrites one of our helpful namespaces (`a`, `m`, etc).  Or maybe they just don't use those and access the motors by name explicitly.  🤷